src/ui/utils/apolloClient.tsx

```diff
@@ -119,8 +119,15 @@ export function extractGraphQLError(error: ApolloError | undefined): string | un
 function createApolloCache() {
   return new InMemoryCache({
     typePolicies: {
+      Query: {
+        fields: {
+          viewer: {
+            merge: true,
+          },
+        },
+      },
       AuthenticatedUser: {
-        keyFields: [],
+        merge: true,
       },
```

Comment:
what's this trying to do?

Reply:
This is interesting! So, Apollo's cache is a flattened, normalized store. This is so that when highly nested objects come in not just the high-level resource is cached, but so are the nested objects. It's a good idea, but it's also an ambitious thing to do, and goes wrong in interesting ways.

By default, the Apollo cache will look for an `id` field in nested resources, and if it finds that it will apply some merging logic to make sure that item gets updated in the store. If it doesn't find that ID key, it blows up. But there are actually many reasonable ways to handle that situation, if you have context on what the objects purpose is. In this case, we know that `AuthenticatedUser` doesn't have an id because there is only ever one of them. This means that anytime we get an AuthenticatedUser in a response, it should overwrite the cache. Adding `merge: true` specifies that behavior. There are lots of other things you can do as well, see the docs for more info: https://www.apollographql.com/docs/react/caching/cache-field-behavior/#the-merge-function